apps/base/utils.py

```diff
@@ -11,3 +15,20 @@ def paginated_queryset(queryset, request):
     paginator.page_size = settings.REST_FRAMEWORK['PAGE_SIZE']
     result_page = paginator.paginate_queryset(queryset, request)
     return (paginator, result_page)
+
+
+@deconstructible
+class RandomFileName(object):
+    def __init__(self, path):
+        self.path = path
+
+    def __call__(self, instance, filename):
+        extension = os.path.splitext(filename)[1]
+        if instance.pk:
```

Comment:
can you please change this if else condition to only include the logic for creating filename. I think apart from creating filename rest two lines can be taken out of if-else condition.